specs/AppLifecycle/Restart/restartApi.markdown

```diff
@@ -0,0 +1,259 @@
+# Restart API
+
+# Contents
+
+- [Restart API](#restart-api)
+- [Contents](#contents)
+- [Background](#background)
+  - [Description](#description)
+- [API](#api)
+  - [Existing API Summary](#existing-api-summary)
+  - [Existing API Details](#existing-api-details)
+  - [New API Details](#new-api-details)
+    - [RequestRestartNow](#requestrestartnow)
+  - [Examples and Scenarios](#examples-and-scenarios)
+- [Telemetry](#telemetry)
+
+# Background
+
+This spec addresses the **Restart APIs** in the AppLifecycle component. At a high-level, Win32 apps can
+register with the OS to restart in update/hang/reboot scenarios but cannot restart immediately on-command 
+with specific arguments/state. This applies to packaged (Desktop Bridge/Centennial) and unpackaged apps. This API will address
+these gaps and provide the ability for any Win32 app to restart immediately.
+
+## Description
+
+The goal of this feature is the new function: **RequestRestartNow**, which will enable any packaged or unpackaged Win32
+app (including Console, WinMain, Windows Forms, WPF and WinUI) to terminate and restart itself on command, and to provide an arbitrary command-line string for the restarted instance.
+
+# API
+
+## Existing API Summary
```

Comment:
Can this also clearly identify why an app developer would want to call the new API instead of the existing APIs? It's telling me what the old APIs do, but not being an expert in this area, I'm not clear why I would want to stop calling the old ones and start calling the new ones.

Reply:
DISCUSSION: There's a servicing gap possible, but there's a race between any restart/servicing operation; there's already an API that does this.  Mostly it's to align WinAppSDK's CoreApp with UWP's features.  Adobe has specifically asked for this win32 app-restart feature, and it's not available (yet) on UWP's CoreApplication.

Reply:
Added a few lines on this.